src/lib/codec/hex/hex.cpp

```diff
@@ -12,31 +12,38 @@
 #include <botan/internal/charset.h>
 #include <botan/internal/ct_utils.h>
 #include <botan/internal/fmt.h>
+#include <botan/internal/loadstor.h>
 
 namespace Botan {
 
 namespace {
 
-char hex_encode_nibble(uint8_t n, bool uppercase) {
-   BOTAN_DEBUG_ASSERT(n <= 15);
-
-   const auto in_09 = CT::Mask<uint8_t>::is_lt(n, 10);
-
-   const char c_09 = n + '0';
-   const char c_af = n + (uppercase ? 'A' : 'a') - 10;
-
-   return in_09.select(c_09, c_af);
+uint16_t hex_encode_2nibble(uint8_t n8, bool uppercase) {
+   auto swar_lt_16 = [](uint16_t a, uint16_t b) -> uint16_t {
+      // This assumes the high bits of both a and b are clear!
+      constexpr uint16_t hi = 0x8080;
+      constexpr uint16_t lo = 0x7F7F;
+      uint16_t r = (lo - a + b) & hi;
+      return (r << 1) - (r >> 7);  // convert to a mask
+   };
+
+   // Offset for upper or lower case 'a' resp
+   const uint16_t a_mask = uppercase ? 0x0707 : 0x2727;
+
+   const uint16_t n = (static_cast<uint16_t>(n8 & 0xF0) << 4) | (n8 & 0x0F);
+   // n >= 10? If so add offset
+   const uint16_t diff = swar_lt_16(0x0909, n) & a_mask;
+   // Can't overflow between bytes, so don't need explicit SWAR addition:
+   return n + 0x3030 + diff;
 }
 
 }  // namespace
 
 void hex_encode(char output[], const uint8_t input[], size_t input_length, bool uppercase) {
    for(size_t i = 0; i != input_length; ++i) {
-      const uint8_t n0 = (input[i] >> 4) & 0xF;
-      const uint8_t n1 = (input[i]) & 0xF;
-
-      output[2 * i] = hex_encode_nibble(n0, uppercase);
-      output[2 * i + 1] = hex_encode_nibble(n1, uppercase);
+      const uint16_t h = hex_encode_2nibble(input[i], uppercase);
+      output[2 * i] = get_byte<0>(h);
+      output[2 * i + 1] = get_byte<1>(h);
```

Comment:
Not sure that's at all easier to read. Mostly due to the cast. :(

```suggestion
      const uint16_t h = hex_encode_2nibble(input[i], uppercase);
      store_be(reinterpret_cast<uint8_t*>(&output[2 * i]), h);
```